src/Microsoft.DncEng.DeployServiceFabricCluster/GatewaySettings.cs

```diff
@@ -0,0 +1,33 @@
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.DncEng.DeployServiceFabricCluster
+{
+    public class GatewaySettings : ResourceGroupDeployerSettings
+    {
+        public Dictionary<string, string>? NeededSecurityGroupRules
+        {
+            get
+            {
+                if (ExternalPorts == null)
+                {
+                    return null;
+                }
+                return new[]
```

Comment:
This... probably need to get broken out a bit. :-) It's a LOT of link and type bouncing back and forth.

I think this is just a new dictionary, and then add the various ports in a loop?

Reply:
Yea, i'll change it.